### DISCUSSION TITLE
celery task

### DISCUSSION THREAD

**Discussion by downdawn** [fastapi-practices/fastapi_best_architecture#237] [General]
Hi，@wu-clan 

There are several questions

### 1. The win system is running abnormally
Solution: [https://github.com/celery/celery/issues/4178](https://github.com/celery/celery/issues/4178)

### 2. Import Exception

When I was testing the task of sending an email, an exception occurred:
```
@celery_app.task
def send_active_email(email: str, active_url: str) -> str:
    send_verification_code_email(to=email, active_url=active_url)
    print(f'发送激活邮件到 {email} 成功')
    return 'Success'
```
KeyError: 'backend.app.tasks.send_active_email'

Request interface: `/api/v1/mixes/tests/send`

KeyError: 'backend.app.tasks.task_demo_async'

```
Received unregistered task of type 'backend.app.tasks.task_demo_async'.
The message has been ignored and discarded.

Did you remember to import the module containing this task?
Or maybe you're using relative imports?

Please see
https://docs.celeryq.dev/en/latest/internals/protocol.html
for more information.

```

### 3. Better control

Asynchronous tasks need to be monitored, you can consider introducing flower
[https://docs.celeryq.dev/en/stable/userguide/monitoring.html#flower-real-time-celery-web-monitor](https://docs.celeryq.dev/en/stable/userguide/monitoring.html#flower-real-time-celery-web-monitor)

### 4. Retry if task fails
At present, task failure will be recorded, but it cannot be retried.
[https://docs.celeryq.dev/en/stable/userguide/tasks.html#retrying](https://docs.celeryq.dev/en/stable/userguide/tasks.html#retrying)
Prioritize retry_backoff

**Comment by wu-clan** [comment]
1. i haven't encountered this problem in Win11 
2. check #238 
3. maybe just library and sh commands, welcome PR 
4. an unknown example? Welcome PR!
